### PR TITLE
feat: new profile page and PATs front-end

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-278/user-profile-page-front-end

Due to the way the work flowed, also includes: https://linear.app/unleash/issue/2-274/frontend-for-personal-api-tokens - I can split it into a separate PR if it's easier to review.

Also includes:
 - TimeAgoCell title refactoring;

### CODE REVIEW

File: frontend/src/component/user/Profile/PasswordTab/PasswordTab.tsx [hunk 1]
```diff
@@ -49,53 +57,30 @@ const EditProfile = ({
...
+                    text: 'Now you can sign in using your new password.',
+                    type: 'success',
+                });
             } catch (error: unknown) {
-                setError(formatUnknownError(error));
+                const formattedError = formatUnknownError(error);
+                setError(formattedError);
+                setToastApiError(formattedError);
             }
         }
```
[the remark this answers is not in the document]
Comment (nunogois) [reply frontend/src/component/user/Profile/PasswordTab/PasswordTab.tsx:76]: @sonatype-lift ignore

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Looks super clean! Minor comments and suggestions. 

From a UI perspective, should the password page use PageContent to get the same appearance as the PAT page? Could also consider this for the Main profile page content. @NicolaeUnleash ?

**Comment by NicolaeUnleash** [comment]
@FredrikOseberg yes the password page should use PageContent to get the same appearance
On the other hand, profile page could be a bit more customized. I will give it a try with Nuno to see how it will look with PageContent, but if it's not working properly we will at least adjust the paddings to align better with the rest

File: frontend/src/component/user/Profile/Profile.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const onChange = (tab: ITab) => {
+        navigate(tab.path ? `/profile/${tab.path}` : '/profile', {
+            replace: true,
+        });
+        setTab(tab.id);
+    };
+
+    const tabFromUrl = () => {
+        const url = location.pathname;
+        const tab = tabs.find(t => t.path && url.includes(t.path));
```
[the remark this answers is not in the document]
Comment (nunogois) [reply frontend/src/component/user/Profile/Profile.tsx:37]: Good catch, bad habit 👍

File: frontend/src/component/user/Profile/PersonalAPITokensTab/CreatePersonalAPIToken/CreatePersonalAPIToken.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    const { setToastApiError } = useToast();
+    const { uiConfig } = useUiConfig();
+    const { locationSettings } = useLocationSettings();
+
+    const [description, setDescription] = useState('');
+    const [expiration, setExpiration] = useState('30d');
+
+    const calculateDate = () => {
+        if (expiration.endsWith('d')) {
+            const days = parseInt(expiration.slice(0, -1));
```
[the remark this answers is not in the document]
Comment (nunogois) [reply frontend/src/component/user/Profile/PersonalAPITokensTab/CreatePersonalAPIToken/CreatePersonalAPIToken.tsx]: Would something like https://github.com/Unleash/unleash/pull/2109/commits/5f89fae7ef8b9267e53bcea577a7f8caacb852b6 help?

[the remark this answers is not in the document]
Comment (FredrikOseberg) [reply frontend/src/component/user/Profile/PersonalAPITokensTab/CreatePersonalAPIToken/CreatePersonalAPIToken.tsx]: Yeah, that looks much more robust to me 👍

**Comment by nunogois** [comment]
Included fixes, refactoring and any other misc changes related to https://linear.app/unleash/issue/2-287/ui-adjustments-for-personal-api-token as well.

**Review by sjaanus — APPROVED** [review]
LGTM!

File: frontend/src/component/user/Profile/PasswordTab/PasswordTab.tsx [hunk 4]
```diff
@@ -49,53 +58,29 @@ const EditProfile = ({
             setError(undefined);
             try {
                 const path = formatApiPath('api/admin/user/change-password');
-                const res = await fetch(path, {
+                await fetch(path, {
                     headers,
                     body: JSON.stringify({ password, confirmPassword }),
                     method: 'POST',
                     credentials: 'include',
                 });
```
Comment (FredrikOseberg) [inline frontend/src/component/user/Profile/PasswordTab/PasswordTab.tsx:66]: This could be moved into an API hook

**Review by FredrikOseberg — APPROVED** [review]
LG, I had one suggestion you could choose to follow or not.

**Review by chriswk — COMMENTED** [review]
Should we use flagResolver to resolve this, the same way we do for embedded proxy?

**Review by chriswk — APPROVED** [review]
I know Fredrik pointed out that there was an interaction that should probably be extracted into a hook. But the logic otherwise seems sound. :+1:

File: frontend/src/component/user/Profile/PasswordTab/PasswordTab.tsx (hunk 4 above)
Comment (nunogois) [reply frontend/src/component/user/Profile/PasswordTab/PasswordTab.tsx]: Addressed in https://github.com/Unleash/unleash/pull/2109/commits/4160084052638f5722c8ef8c28dbb0c753cfd89b

**Review by FredrikOseberg — APPROVED** [review]

**Pull request merged**